### ISSUE TITLE
~master dependencies should be upgraded automatically

### ISSUE THREAD

**Issue by MartinNowak** [dlang/dub#131]
Pretty simple issue, once the master package is installed it never gets updated. Only reinstalling helps currently.

**Comment by WebFreak001** [comment]
branch based dependencies have been deprecated / discouraged for a while for this reason.

I think to go forward with this, like described in #9, we need:

- some checksum in dub.selections.json (#1723)
- registry keeping track of checksums of repo / branch dependencies

**Comment by rikkimax** [comment]
Wouldn't we use the commit id of the VCS repo rather than a checksum?

**Comment by WebFreak001** [comment]
yes, that's actually a better way how to keep track of git dependencies, but we would probably still want the registry to send people what's the current version, so locally it can be checked if an update is needed.

**Comment by ntrel** [comment]
We would also need a way to specify a git branch for `dub run`. E.g. currently this works (though master is only fetched the first time):

	dub run dlang-tour@~master -- --lang-dir .

**Comment by WebFreak001** [comment]
@ntrel please open a new issue for that, that's unrelated to this issue
